v2/tools/generator/internal/codegen/pipeline/handle_float_values.go

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package pipeline
+
+import (
+	"context"
+
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/astmodel"
+	"github.com/pkg/errors"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+)
+
+// HandleFloatValuesStageID is the unique identifier of this stage
+const (
+	HandleFloatValuesStageID = "handleFloatValues"
+)
+
+func HandleFloatValues() *Stage {
+	stage := NewStage(
+		HandleFloatValuesStageID,
+		"Transform Float type values to Integer types",
+		func(ctx context.Context, state *State) (*State, error) {
+
+			definitions := state.Definitions()
+			specDefinitions := astmodel.FindSpecDefinitions(definitions)
+
+			result, errs := getFloatTransformations(definitions, specDefinitions)
+			if err := kerrors.NewAggregate(errs); err != nil {
+				return nil, err
+			}
+
+			remaining := definitions.Except(result)
+			result.AddTypes(remaining)
+
+			return state.WithDefinitions(result), nil
+		})
+
+	stage.RequiresPrerequisiteStages(RemoveStatusPropertyValidationsStageID)
+
+	return stage
+}
+
+func getFloatTransformations(definitions astmodel.TypeDefinitionSet, validatedDefinitions astmodel.TypeDefinitionSet) (astmodel.TypeDefinitionSet, []error) {
+	walker := astmodel.NewTypeWalker(
+		definitions,
+		astmodel.TypeVisitorBuilder{
+			VisitObjectType: transformFloatToInt,
+		}.Build())
+
+	return walkTypes(validatedDefinitions, walker)
+}
+
+// transformFloatToInt transforms all the FloatTypes to IntergerTypes
+func transformFloatToInt(this *astmodel.TypeVisitor, ot *astmodel.ObjectType, ctx interface{}) (astmodel.Type, error) {
+	for _, prop := range ot.Properties() {
+		// check if the base type is Float64 Type
+		if result, ok := astmodel.AsPrimitiveType(prop.PropertyType()); ok && result == astmodel.FloatType {
```

Comment:
The `AsX()` methods are great for querying types but shouldn't be used for transforming them - you end up throwing away any/all wrappers around the types (these properties might have `OptionalType`, `FlaggedType`, etc around them; you want to change the interior of the onion, not blow it away).

The TypeVisitor can do this iteration/unwrap/rewrap for you: 
* Add a handler for `VisitValidatedType` that passes in a context object to the recursive call
* Add a handler for `VisitPrimitive` that checks for context + float and changes returns an int instead